Network/NetworkService.swift

```diff
@@ -2,39 +2,45 @@ import Foundation
 
 class NetworkService {
     
-    private var task: NSURLSessionDataTask?
-    private var successCodes: Range<Int> = 200..<299
-    private var failureCodes: Range<Int> = 400..<499
+    private var task: URLSessionDataTask?
+    private var successCodes: CountableRange<Int> = 200..<299
+    private var failureCodes: CountableRange<Int> = 400..<499
     
     enum Method: String {
-        case GET, POST, PUT, DELETE
+        case get, post, put, delete
     }
     
-    func request(url url: NSURL, method: Method,
-                 params: [String: AnyObject]? = nil,
+    enum QueryType {
+        case json, path
+    }
+    
+    func makeRequest(for url: URL, method: Method, query type: QueryType?,
+                 params: [String: Any]? = nil,
                  headers: [String: String]? = nil,
-                 success: (NSData? -> Void)? = nil,
-                 failure: ((data: NSData?, error: NSError?, responseCode: Int) -> Void)? = nil) {
+                 success: ((Data?) -> Void)? = nil,
+                 failure: ((_ data: Data?, _ error: NSError?, _ responseCode: Int) -> Void)? = nil) {
         
-        let mutableRequest = NSMutableURLRequest(URL: url, cachePolicy: .ReloadIgnoringLocalAndRemoteCacheData,
-                                                 timeoutInterval: 10.0)
-        mutableRequest.allHTTPHeaderFields = headers
-        mutableRequest.HTTPMethod = method.rawValue
-        if let params = params {
-            mutableRequest.HTTPBody = try! NSJSONSerialization.dataWithJSONObject(params, options: [])
+        guard let params = params, let type = type else {
```

Comment:
I do not understand this one. Probably no more valid. We always should have a `type` here because it is non-optional, so this is clearly bug to me (right?). The next thing is that the parameters are never required, they might be optional and I believe that `makeQuery` method should know how to deal with request if there is no params to be set.